### PR TITLE
Overlay changes

### PR DESCRIPTION
**Pull request by EperezOk**

### CODE REVIEW

File: contracts/src/MerkleClaimERC20.sol [hunk 1]
```diff
@@ -1,79 +1,80 @@
...
+    /// @param amount of tokens owed to claimee
+    /// @param proof merkle proof to prove address and amount are in tree
+    function claim(address to, uint256 amount, bytes32[] calldata proof) external {
+        // Throw if address has already claimed tokens
+        if (hasClaimed[to]) revert AlreadyClaimed();
 
-    // Mint tokens to address
-    _mint(to, amount);
+        // Throw if the contract doesn't hold enough tokens for claimee
+        if (amount > token.balanceOf(address(this))) revert NotEnoughRewards();
```
Comment (TomasCImach) [inline contracts/src/MerkleClaimERC20.sol:64]: This wouldn't be neccessary, in this case the tx will revert when `token.transfer(to,amount)` is called.

File: contracts/src/OverlayV1Token.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    function mint(address _recipient, uint256 _amount) external onlyMinter {
+        _mint(_recipient, _amount);
+    }
+
+    function burn(uint256 _amount) external onlyBurner {
+        _burn(msg.sender, _amount);
+    }
+
+    function unlockTransfers() external onlyRole(DEFAULT_ADMIN_ROLE) {
```
Comment (TomasCImach) [inline contracts/src/OverlayV1Token.sol]: revert to previous ovl token version with no transfer locks

File: contracts/src/test/utils/MerkleClaimERC20Test.sol [hunk 3]
```diff
@@ -3,38 +3,43 @@ pragma solidity >=0.8.0;
...
+
+        // Create airdrop token
+        TOKEN = new MerkleClaimERC20(
+            ovl,
+            // Merkle root containing ALICE with 100e18 tokens but no BOB
+            0xd0aa6a4e5b4e13462921d7518eebdb7b297a7877d6cfe078b0c318827392fb55
+        );
+
+        ovl.grantRole(TRANSFER_ROLE, address(TOKEN)); // allow contract to transfer OVL
+        deal(address(ovl), address(TOKEN), 100e18); // total amount of tokens to be claimed
```
Comment (TomasCImach) [inline contracts/src/test/utils/MerkleClaimERC20Test.sol:37]: Setup by transfer the token to the SC, as it'll be done when deployed.

```suggestion
        deal(address(this), address(TOKEN), 100e18); // total amount of tokens to be claimed
        TOKEN.transfer(address(ovl), 100e18);
```

File: contracts/src/test/utils/MerkleClaimERC20Test.sol [hunk 4]
```diff
@@ -3,38 +3,43 @@ pragma solidity >=0.8.0;
...
+    /// @dev User: Bob (not in merkle tree)
+    MerkleClaimERC20User internal BOB = MerkleClaimERC20User(0xEFc56627233b02eA95bAE7e19F648d7DcD5Bb132);
+
+    /// ============ Setup test suite ============
+
+    function setUp() public virtual {
+        OverlayV1Token ovl = new OverlayV1Token();
+
+        // Create airdrop token
+        TOKEN = new MerkleClaimERC20(
```
Comment (TomasCImach) [inline contracts/src/test/utils/MerkleClaimERC20Test.sol:31]: Change variable name, as this is no longer a token.

File: generator/.prettierrc.json
Comment (TomasCImach) [inline generator/.prettierrc.json:1]: Refrain from altering formats within the same pull request where primary changes are made.

**Review by TomasCImach — CHANGES_REQUESTED** [review]
Is it possible to move the commits of the list generation to another branch/PR?

File: contracts/src/MerkleClaimERC20.sol (hunk 1 above)
Comment (EperezOk) [reply contracts/src/MerkleClaimERC20.sol:64]: I borrowed it from the [Frax finance implementation](https://github.com/FraxFinance/merkle-airdrop-starter/blob/master/contracts/src/MerkleClaim.sol#L80-L81), but we can remove it

**Comment by EperezOk** [comment]
Changes in this PR will be split into #2 and #3.

**Pull request closed without merging**